test/e2e/set/scyllacluster/verify.go

```diff
@@ -116,13 +118,32 @@ func verifyScyllaCluster(ctx context.Context, kubeClient kubernetes.Interface, s
 
 	verifyPersistentVolumeClaims(ctx, kubeClient.CoreV1(), sc)
 
-	// TODO: Use scylla client to check at least "UN"
 	scyllaClient, hosts, err := utils.GetScyllaClient(ctx, kubeClient.CoreV1(), sc)
 	o.Expect(err).NotTo(o.HaveOccurred())
 	defer scyllaClient.Close()
 
 	o.Expect(hosts).To(o.HaveLen(memberCount))
 
+	// Because of Gossip issues node status is not consistent across different scylla nodes.
+	// Until it's fixed we need to wait for all the nodes to sync to make sure the CQL doesn't hit issues later on.
+	// This loop should be eventually removed and only the explicit and immediate check on a single node should stay.
+	o.Eventually(func(eo o.Gomega) {
+		statuses := make([]scyllaclient.NodeStatusInfoSlice, 0, len(hosts))
+		for _, h := range hosts {
+			s, err := scyllaClient.Status(ctx, h)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			framework.Infof("Node %q, down: %q, up: %q", h, strings.Join(s.DownHosts(), ","), strings.Join(s.LiveHosts(), ","))
+			statuses = append(statuses, s)
+		}
+
+		for _, s := range statuses {
+			eo.Expect(s.DownHosts()).To(o.BeEmpty())
+			eo.Expect(s.LiveHosts()).To(o.HaveLen(memberCount))
+			eo.Expect(s.Hosts()).To(o.HaveLen(memberCount))
+		}
+	}).WithTimeout(60 * time.Second).WithPolling(1 * time.Second).Should(o.Succeed())
```

Comment:
What's the source of the 60 seconds? I'd say it needs a comment explaining why we think it's enough time for the gossip to propagate.